api/v1alpha1/perses_types.go

```diff
@@ -90,11 +90,49 @@ type PersesService struct {
 }
 
 type Client struct {
+	// BasicAuth basic auth config for datasource client
+	// +optional
+	BasicAuth *BasicAuth `json:"basicAuth,omitempty"`
```

Comment:
Do you think we should use this interfaces from the perses API rather that redefining them here?

Reply:
Well, all the types are slightly different than the perses API due to the `SecretSourceType` needed to pull the data from different sources, unless you mean in a different spot?